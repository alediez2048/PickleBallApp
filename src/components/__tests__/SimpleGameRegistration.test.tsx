import React from 'react';
import { Text, View, TouchableOpacity } from 'react-native';
import renderer from 'react-test-renderer';

// Simple sample component without any hooks or context
const SimpleGameRegistration = (props: { spotsLeft: number; isLoading: boolean; isFull: boolean }) => {
  const { spotsLeft, isLoading, isFull } = props;
  
  const formatSpotsMessage = () => {
    if (isLoading) return 'Loading...';
    if (isFull) return 'Game Full';
    return `${spotsLeft}/12 spots left`;
  };

  return (
    <View testID="game-registration">
      <Text testID="spots-message">{formatSpotsMessage()}</Text>
      {isLoading && <Text testID="loading-indicator">Loading...</Text>}
      {isFull && <Text testID="full-message">Game is full</Text>}
      <TouchableOpacity 
        testID="register-button"
        disabled={isFull}
        onPress={() => {}}
      >
        <Text>Register</Text>
      </TouchableOpacity>
    </View>
  );
};

describe('SimpleGameRegistration', () => {
  it('renders correctly with various prop combinations', () => {
    // Available spots
    const availableSpotsTree = renderer
      .create(<SimpleGameRegistration spotsLeft={7} isLoading={false} isFull={false} />)
      .toJSON();
    expect(availableSpotsTree).toMatchSnapshot();
    
    // Loading state
    const loadingTree = renderer
      .create(<SimpleGameRegistration spotsLeft={0} isLoading={true} isFull={false} />)
      .toJSON();
    expect(loadingTree).toMatchSnapshot();
    
    // Full game
    const fullGameTree = renderer
      .create(<SimpleGameRegistration spotsLeft={0} isLoading={false} isFull={true} />)
      .toJSON();
    expect(fullGameTree).toMatchSnapshot();
  });
}); 