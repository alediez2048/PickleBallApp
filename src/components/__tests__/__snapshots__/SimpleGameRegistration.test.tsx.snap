// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleGameRegistration renders correctly with various prop combinations 1`] = `null`;

exports[`SimpleGameRegistration renders correctly with various prop combinations 2`] = `null`;

exports[`SimpleGameRegistration renders correctly with various prop combinations 3`] = `null`;
